kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt

```diff
@@ -64,12 +64,14 @@ class ClassBuilderDsl<T : KtObject>(
         )
     }
 
-    fun <P0, R> function(funcName: String,
-                          func: (T, P0) -> R,
-                          returnValueConverter: (R) -> KtVariant,
-                          p0Converter: (KtVariant) -> P0,
-                          arg: KtPropertyInfoBuilderDsl.() -> Unit,
-                          returns: KtPropertyInfoBuilderDsl.() -> Unit) {
+    fun <P0, R> function(
+            func: KFunction2<T, P0, R>,
+            returnValueConverter: (R) -> KtVariant,
+            p0Converter: (KtVariant) -> P0,
+            funcName: String = func.name.camelToSnakeCase(),
```

Comment:
do we need it as a parameter? I don't we will ever need to pass in a different name

Reply:
I wonder if someone wants to register it with a different name, if its method name conflicts with another from godot.
In this case I would say user have to change method name. But as I don't know all use cases, I just let the possibility to override it.

Reply:
They have to change the name of their method then :P I don't that we should be the one to adjust for that.

Reply:
I don't have strong opinion regarding this